packages/framework/src/Component/Grid/GridView.php

```diff
@@ -149,23 +149,42 @@ public function renderCell(Column $column, ?array $row = null, ?FormView $formVi
     }
 
     /**
-     * @param \Shopsys\FrameworkBundle\Component\Grid\ActionColumn $actionColumn
+     * @param \Shopsys\FrameworkBundle\Component\Grid\GridRowColumnInterface $actionColumn
      * @param array $row
      */
-    public function renderActionCell(ActionColumn $actionColumn, array $row): void
+    public function renderActionCell(GridRowColumnInterface $actionColumn, array $row): void
     {
-        $possibleBlocks = [
-            'grid_action_cell_type_' . $actionColumn->getType(),
-            'grid_action_cell',
-        ];
+        if ($actionColumn instanceof ActionColumn) {
+            $possibleBlocks = [
+                'grid_action_cell_type_' . $actionColumn->getType(),
+                'grid_action_cell',
+            ];
 
-        foreach ($possibleBlocks as $blockName) {
-            if ($this->blockExists($blockName)) {
-                $this->renderBlock($blockName, ['actionColumn' => $actionColumn, 'row' => $row]);
+            foreach ($possibleBlocks as $blockName) {
+                if ($this->blockExists($blockName)) {
+                    $this->renderBlock($blockName, ['actionColumn' => $actionColumn, 'row' => $row]);
 
-                break;
+                    break;
+                }
             }
+
+            return;
         }
+
+        if ($actionColumn instanceof GridRowActionInterface) {
+            if ($actionColumn->validate($row) === false) {
+                return;
+            }
+
+            $renderData = $actionColumn->renderData();
+            echo $this->twig->render($renderData['template'], [...$renderData['parameters'], 'row' => $row]);
+
+            return;
+        }
+
+        throw new InvalidArgumentException(
```

Comment:
this is quite weird - you declare `GridRowColumnInterface` can be passed to `renderActionCell`, however, only `ActionColumn`|`GridRowActionInterface` are actually supported. Could we discuss it, please?

Reply:
refactored :)